src/components/BaseButton.vue

```diff
@@ -0,0 +1,31 @@
+<template>
+ <button type="button" :class="size" :style="{'background-color':backgroundColor}" >Trimite
+     <slot/>
+  </button>
+</template>
+
+<script>
+export default {
+  name: 'BaseButton',
+  props: {
+    size: {
+        type: String,
+        default: 'base-button'
+     },
+     backgroundColor: {
+       type: String,
+     }
+    }
+  }
+</script>
+
+<style >
+@baseButton component {
+  .base-button {
+    @apply inline-flex items-center px-4 py-2 border border-transparent text-base font-medium rounded-md shadow-sm text-white bg-indigo-600 active:bg-indigo-800 disabled:opacity-50;
```

Comment:
when you have classes that are repeating please use a single class name .button and only add the different classes for different types of buttons

Reply:
All the classes that are repeating, can be removed, and left only on the button class,
 @apply inline-flex items-center border border-transparent font-medium rounded-md shadow-sm text-white bg-indigo-600 active:bg-indigo-800 disabled:opacity-50